Wrappers/Python/ccpi/optimisation/operators/ChannelwiseOperator.py

```diff
@@ -0,0 +1,184 @@
+        elif isinstance(dom_op, AcquisitionGeometry):
+            d = AcquisitionGeometry(
+                                   dom_op.geom_type,
+                                   dom_op.dimension, 
+                                   dom_op.angles, 
+                                   dom_op.pixel_num_h, 
+                                   dom_op.pixel_size_h, 
+                                   dom_op.pixel_num_v, 
+                                   dom_op.pixel_size_v, 
+                                   dom_op.dist_source_center, 
+                                   dom_op.dist_center_detector, 
+                                   channels,
+                                   dimension_labels=[channel_label] + dom_op.dimension_labels)
+        else:
+            pass
+        
+        if isinstance(ran_op, ImageGeometry):
+            r = ImageGeometry(  
+                            ran_op.voxel_num_x, 
+                            ran_op.voxel_num_y, 
+                            ran_op.voxel_num_z, 
+                            ran_op.voxel_size_x, 
+                            ran_op.voxel_size_y, 
+                            ran_op.voxel_size_z, 
+                            ran_op.center_x, 
+                            ran_op.center_y, 
+                            ran_op.center_z, 
+                            channels,
+                            dimension_labels=[channel_label] + ran_op.dimension_labels)
+        elif isinstance(ran_op, AcquisitionGeometry):
+            r = AcquisitionGeometry(
+                                   ran_op.geom_type,
+                                   ran_op.dimension, 
+                                   ran_op.angles, 
+                                   ran_op.pixel_num_h, 
+                                   ran_op.pixel_size_h, 
+                                   ran_op.pixel_num_v, 
+                                   ran_op.pixel_size_v, 
+                                   ran_op.dist_source_center, 
+                                   ran_op.dist_center_detector, 
+                                   channels,
+                                   dimension_labels=[channel_label] + ran_op.dimension_labels)
+        else:
+            pass
+        
+        super(ChannelwiseOperator, self).__init__(domain_geometry=d, 
+                                           range_geometry=r)
+        
+        self.op = op
+        self.channels = channels
+        self.channel_label = channel_label
+
+        
+    def direct(self,x,out=None):
+        
+        '''Returns D(x)'''
+        
+        # Initialise output
+        output = self.range_geometry().allocate()
+        output_array = output.as_array()
+        cury = self.op.range_geometry().allocate()
+        
+        for k in range(self.channels):
+            curx = x.subset(channel=k)
+            self.op.direct(curx,cury)
+            output_array[k] = cury.as_array()
```

Comment:
Ok, fill sounds promising. Do you mean dimension='channel' rather than dimension='angle'? Also how would one specify that one want cury to be filled into channel no k?

Reply:
Tried this, it runs, but it feels ALL channels with the cury.as_array(), rather than just the `k`th one as desired. This happens both with dimension='angle' and ='channel'. Is there a way to fill on channel `k`?

Reply:
oh yes. then could you check that this works as expected?

```python

output.fill(cury.as_array(), channel=k)
```

Reply:
YES! That does exactly that! Brilliant, thanks!  What a fantastic method, this fill!  I'll update the code with this.  Any ideas for the Q1 on whether there's a more elegant to create a multichannel geometry from a single-channel one?